### PR TITLE
[#139291295] fix lint_terraform

### PR DESCRIPTION
**Pull request by mtekel**
## What

[Fix `make lint_terraform`](https://www.pivotaltracker.com/n/projects/1275640/stories/139291295)

We now have TF 0.8.5 and a new version of provider built for this TF. Use that for lint checks.

## How to review

`make lint_terraform` should succeed.

## Who can review

not @mtekel or @saliceti

### CODE REVIEW

**Comment by alext** [comment]
Looks good to me.

I wonder whether it's mentioning in a commit message that the reason we need to download the provider is that with TF 0.8 it's now needed for the `terraform graph` and `terraform fmt` commands, whereas it wasn't for earlier versions.

**Comment by mtekel** [comment]
I've updated the commit message to mention that provider is required by TF 0.8.5. to do graph.

File: Makefile [hunk 1]
```diff
@@ -44,9 +44,10 @@ lint_yaml:
 	find . -name '*.yml' -not -path '*/vendor/*' | xargs $(YAMLLINT) -c yamllint.yml
 
 .PHONY: lint_terraform
-lint_terraform: dev ## Lint the terraform files.
+lint_terraform: check-tf-version dev ## Lint the terraform files.
```
Comment (dcarley) [inline Makefile]: 👍  I now realise why it wasn't failing for me - I had Terraform 0.7 locally.

**Comment by dcarley** [comment]
LGTM. Travis failed for an unrelated reason. I've restarted the build, but we might need to raise the timeout for this test if we see it fail regularly:
```
• Failure [1.004 seconds]
UpdateDataDogMonitor
/home/travis/build/alphagov/paas-cf/concourse/scripts/update_monitor_options_test.go:260
  when API responds with non 200 code to PUT monitor
  /home/travis/build/alphagov/paas-cf/concourse/scripts/update_monitor_options_test.go:259
    exits with non 0 code [It]
    /home/travis/build/alphagov/paas-cf/concourse/scripts/update_monitor_options_test.go:258
    Timed out after 1.001s.
    Expected process to exit.  It did not.
    /home/travis/build/alphagov/paas-cf/concourse/scripts/update_monitor_options_test.go:257
------------------------------
•••••••••••••••••••••••••••••••
Summarizing 1 Failure:
[Fail] UpdateDataDogMonitor when API responds with non 200 code to PUT monitor [It] exits with non 0 code 
/home/travis/build/alphagov/paas-cf/concourse/scripts/update_monitor_options_test.go:257
Ran 40 of 40 Specs in 11.086 seconds
FAIL! -- 39 Passed | 1 Failed | 0 Pending | 0 Skipped --- FAIL: TestScripts (11.09s)
FAIL
exit status 1
FAIL	_/home/travis/build/alphagov/paas-cf/concourse/scripts	11.176s
make: *** [spec] Error 1
The command "make test" exited with 2.
Done. Your build exited with 1.
```

**Pull request merged**